features/barebone_tests.feature

```diff
@@ -5,17 +5,16 @@ Feature: Barebone tests
 
   Scenario: Barebone test: handled errors
     When I run "BareboneTestHandledScenario"
-    And I wait to receive 3 requests
-  
-    Then the request is valid for the session reporting API version "1.0" for the "iOS Bugsnag Notifier" notifier
-    And the payload field "sessions.0.id" is not null
+    And I wait to receive a session
+    And I wait to receive 2 errors
+
+    Then the session is valid for the session reporting API version "1.0" for the "iOS Bugsnag Notifier" notifier
+    And the session payload field "sessions.0.id" is not null
     And the session "user.id" equals "foobar"
     And the session "user.email" equals "foobar@example.com"
     And the session "user.name" equals "Foo Bar"
 
-    And I discard the oldest request
-
-    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
+    Then the error is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
```

Comment:
It may be good down the line to create steps that will run this in bulk across all received errors or sessions, I'd suggest a separate ticket for this however.

Reply:
Yep, completely agreed and Nick made the same observation - PLAT-5782 will cover it.